platform/features/table/src/controllers/HistoricalTableController.js

```diff
@@ -35,32 +36,79 @@ define(
             var rowData = [],
-                self = this;
+                self = this,
+                telemetryObjects = this.handle.getTelemetryObjects();
 
-            this.handle.getTelemetryObjects().forEach(function (telemetryObject) {
-                var series = self.handle.getSeries(telemetryObject) || {},
-                    pointCount = series.getPointCount ? series.getPointCount() : 0,
-                    i = 0;
+            function processTelemetryObject(offset) {
+                var telemetryObject = telemetryObjects[offset],
+                    series = self.handle.getSeries(telemetryObject) || {},
+                    pointCount = series.getPointCount ? series.getPointCount() : 0;
 
-                for (; i < pointCount; i++) {
-                    rowData.push(self.table.getRowValues(telemetryObject,
-                        self.handle.makeDatum(telemetryObject, series, i)));
+                function processBatch(start, end) {
+                    var i;
+                    end = Math.min(pointCount, end);
+
+                    clearTimeout(self.timeoutHandle);
+                    delete self.timeoutHandle;
+
+                    //The row offset (ie. batch start point) does not exceed the rows available
+                    for (i = start; i < end; i++) {
+                        rowData.push(self.table.getRowValues(telemetryObject,
+                            self.handle.makeDatum(telemetryObject, series, i)));
+                    }
+                    if (end < pointCount) {
+                        //Yield if necessary
+                        self.timeoutHandle = setTimeout(function () {
+                            processBatch(end, end + self.batchSize);
+                        }, 0);
+                    } else {
+                        //All rows for this object have been processed, so check if there are more objects to process
+                        offset++;
+                        if (offset < telemetryObjects.length) {
+                            //More telemetry object to process
+                            processTelemetryObject(offset);
+                        } else {
+                            // No more objects to process. Apply rows to scope
+                            // Apply digest. Digest may be in progress (if batch small
+                            // enough to not require yield), so using $timeout instead
+                            // of $scope.$apply to avoid in progress error
+                            self.$timeout(function () {
+                                self.$scope.loading = false;
+                                self.$scope.rows = rowData;
+                            });
+                        }
+                    }
                 }
-            });
+                processBatch(0, self.batchSize);
+            }
+
+            if (telemetryObjects.length > 0) {
+                this.$scope.loading = true;
```

Comment:
Yeah, agreed. The reason for having it here was just to avoid setting it in one class (`TableController`) and unsetting it in another (`HistoricalTableController`). I'll change it so its set at request time, add a comment to make it clear that it's unset elsewhere.